.github/workflows/build-test-windows.yml

```
@@ -2,11 +2,23 @@
 name: build-and-test-windows
 on:
   push:
-    branches: [ main, master ]
+    branches:
+      - master
+
+    paths-ignore:
+      - '.github/**'
+      - '!.github/workflows/build.yml'
+      - '!.github/workflows/build-test-windows.yml'
+      - '**.md'
+
   pull_request:
     types: [opened, synchronize, reopened, ready_for_review]
   workflow_dispatch:
 
+concurrency:
+  group: build-and-test-windows-${{ github.ref_name }}
```

I don't think that we need to write the literal string `build-and-test-windows` here. We should be able to use a placeholder like you did with `${{ github.ref_name}}`.

I think this should be
```suggestion
  group: ${{ github.workflow }}-${{ github.ref_name }}
```

---

Reasonable. Fixed it with the newest commit. Nice catch on this one though.